docs/modules/cloudhub/CloudHub.md

```diff
@@ -0,0 +1,57 @@
+# CloudHub
+
+## Overview
+
+Cloud hub is a web socket client, which is mediator between EdgeController and the Edge side.
+It function is to do the communication between edge and the EdgeController.
+
+The connection to the edge(through EdgeHub module) is done through the HTTP over Websocket connection.
+For internal communication it directly communicates with the EdgeController.
+All the request send to CloudHub is of context object which are stored in channelQ along with the
+mapped channels of event object marked to its nodeID.
+
+
+The main functions performed by CloudHub are :-
+
+- Get message context and create ChannelQ for events
+- Create http connection over websocket
+- Serving Websocket connection
+- Reading message from edge
+- Writing message to edge
+- Publish message to Controller
+
+
+## Get message context and create ChannelQ for events
```

Comment:
In readthedocs all headings with ## get added as a menu item in the table of contents and the sidebar menu.  
Any Heading that must be part of the menu should be titled with ##.  
Else make it a subtitle using ###